include/deal.II/base/polynomials_hermite.h

```diff
@@ -0,0 +1,158 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+#ifndef dealii_hermite_polynomials
+#define dealii_hermite_polynomials
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/subscriptor.h>
+
+#include <deal.II/lac/full_matrix.h>
+
+#include <algorithm>
+#include <exception>
+#include <memory>
+#include <vector>
+
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * @addtogroup Polynomials
+ * @{
+ */
+
+namespace Polynomials
+{
+  /**
+   * This class implements Hermite interpolation polynomials (see 
+   * @cite CiarletRiavart1972interpolation) enforcing the maximum 
+   * posible level of regularity $r$ in the FEM basis given a 
+   * polynomial degree of $2r+1$. The polynomials all represent 
+   * either a non-zero shape value or derivative at $y=0$ and $y=1$ 
+   * on the reference interval $y \in [0,1]$.
+   *
+   * Indices $j = 0, 1, \dots, r$ refer to polynomials corresponding
+   * to a non-zero derivative (or shape value for $j=0$) of
+   * order $j$ at $x=0$, and indices $j = r+1, r+2, \dots, 2r+1$
+   * refer to polynomials with a non-zero derivative of order
+   * $j-(r+1)$ (or value for $j=r+1$) at $x=1$. The basis is rescaled
+   * so that a function corresponding to a non-zero $j^{th}$
+   * derivative has derivative value $j! 4^{j}$ at the corresponding
+   * node.
+   */
+  class HermitePoly : public Polynomial<double>
```

Comment:
All of the other classes in this direction have names of the form `PolynomialsX`. So maybe call this one `PolynomialsHermite`?